### PR TITLE
remove BC-breaking changes

### PR DESCRIPTION
**Pull request by eellison**
`IntermediateLayerGetter` and `DenseBlock` were re-written to be supported in script, along the way introducing BC changes that required `_load_from_state_dict` logic. 

This removes those changes, moving them to `ModuleDict`. Once https://github.com/pytorch/pytorch/pull/28988 lands we can re-enable these scripting tests.

### CODE REVIEW

File: test/test_models.py [hunk 1]
```diff
@@ -38,13 +38,17 @@ def get_available_video_models():
 # models that are in torch hub, as well as r3d_18. we tried testing all models
 # but the test was too slow. not included are detection models, because
 # they are not yet supported in JIT.
+
+# Temp disabled modules: disabled until https://github.com/pytorch/pytorch/pull/28988 lands
+# so that we do not have to introduce BC changes.
+
 script_test_models = [
-    "deeplabv3_resnet101",
+    # "deeplabv3_resnet101", temp disabled
```
Comment (driazati) [inline test/test_models.py]: If this is all going in before a release it's better to just wait until pytorch#28988 lands then land this PR so these tests never have to be disabled

Comment (eellison) [reply test/test_models.py]: AFAIK There's a minor release of torchvision coming up, this PR is for that release

Comment (fmassa) [reply test/test_models.py]: The minor release of torchvision didn't contain the BC-breaking commits, so I think we can wait until pytorch#28988 lands to merge this PR, thoughts?

Comment (eellison) [reply test/test_models.py]: Okay cool, in that case yes we can wait.

Comment (fmassa) [reply test/test_models.py]: @eellison https://github.com/pytorch/pytorch/pull/28988 has landed, can you re-enable this test?

**Comment by fmassa** [comment]
Merging this as the failures are unrelated.

**Pull request merged**